diff --git a/llarp/config/config.cpp b/llarp/config/config.cpp
index 78d152602..8b07b0cec 100644
--- a/llarp/config/config.cpp
+++ b/llarp/config/config.cpp
@@ -703,7 +703,10 @@ namespace llarp
     // can bind to other 127.* IPs to avoid conflicting with something else that may be listening on
     // 127.0.0.1:53.
 #ifdef __linux__
-    constexpr Default DefaultDNSBind{"127.3.2.1:53"};
+    // Fedora's systemd-resolved seems unable to connect to 127.3.2.1 for unknown reasons,
+    // however since systemd-resolved is perfectly happy with a different port so listen on
+    // localhost:953 as a workaround.
+    constexpr Default DefaultDNSBind{"127.0.0.1:953"};
 #else
     constexpr Default DefaultDNSBind{"127.0.0.1:53"};
 #endif
