From cdad3e7f093c4b0c69f73580e4fbacc24067db96 Mon Sep 17 00:00:00 2001
From: necro-nemsis <necro_nemesis@hotmail.com>
Date: Sun, 19 Jun 2022 06:47:01 -0400
Subject: [PATCH] Change default port to 953

---
 llarp/config/config.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/llarp/config/config.cpp b/llarp/config/config.cpp
index 6bff9611..9c89bce6 100644
--- a/llarp/config/config.cpp
+++ b/llarp/config/config.cpp
@@ -760,7 +760,10 @@ namespace llarp
     // can bind to other 127.* IPs to avoid conflicting with something else that may be listening on
     // 127.0.0.1:53.
 #ifdef __linux__
-    constexpr Default DefaultDNSBind{"127.3.2.1:53"};
+    // Fedora's systemd-resolved seems unable to connect to 127.3.2.1 for unknown reasons,
+    // however since systemd-resolved is perfectly happy with a different port so listen on
+    // localhost:953 as a workaround.
+    constexpr Default DefaultDNSBind{"127.3.2.1:953"};
 #else
     constexpr Default DefaultDNSBind{"127.0.0.1:53"};
 #endif
-- 
2.30.2

