From: Jason Rhinelander <jason@imaginary.ca>
Date: Mon, 30 May 2022 21:56:50 -0300
Subject: oxenmq linking fix

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bf5e6ca..d884a1e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -200,7 +200,9 @@ if(NOT FORCE_OXENMQ_SUBMODULE)
   pkg_check_modules(OXENMQ liboxenmq>=1.2.12 IMPORTED_TARGET)
 endif()
 if(OXENMQ_FOUND)
-  add_library(oxenmq::oxenmq ALIAS PkgConfig::OXENMQ)
+  add_library(oxenmq INTERFACE)
+  target_link_libraries(oxenmq INTERFACE PkgConfig::OXENMQ)
+  add_library(oxenmq::oxenmq ALIAS oxenmq)
   message(STATUS "Found system liboxenmq ${OXENMQ_VERSION}")
 else()
   message(STATUS "using oxenmq submodule")
