From: Jeff Becker <jeff@i2p.rocks>
Date: Fri, 11 Jun 2021 11:47:24 -0400
Subject: use exisitng convotag first instead of trying to send to directly

---
 llarp/handlers/tun.cpp | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/llarp/handlers/tun.cpp b/llarp/handlers/tun.cpp
index c816297..f9e2820 100644
--- a/llarp/handlers/tun.cpp
+++ b/llarp/handlers/tun.cpp
@@ -973,10 +973,13 @@ namespace llarp
         }
         // try sending it on an existing convotag
         // this succeds for inbound convos, probably.
-        if (SendToOrQueue(to, pkt.ConstBuffer(), type))
+        if (auto maybe = GetBestConvoTagFor(to))
         {
-          MarkIPActive(dst);
-          return;
+          if (SendToOrQueue(*maybe, pkt.ConstBuffer(), type))
+          {
+            MarkIPActive(dst);
+            return;
+          }
         }
         // try establishing a path to this guy
         // will fail if it's an inbound convo
