From: Jason Rhinelander <jason@imaginary.ca>
Date: Tue, 10 Sep 2019 19:23:47 -0300
Subject: Add no-resolvconf hack to default config

---
 llarp/config/config.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/llarp/config/config.cpp b/llarp/config/config.cpp
index 87589c7..662d68d 100644
--- a/llarp/config/config.cpp
+++ b/llarp/config/config.cpp
@@ -654,6 +654,11 @@ llarp_generic_ensure_config(std::ofstream &f, std::string basepath,
   f << "bind=127.0.0.1:1153\n";
 #else
   f << "bind=127.3.2.1:53\n";
+
+  f << "# Uncomment if you want to disable trying to call resolvconf to update the system nameserver\n"
+    << "# (only applies if the system is actually using resolvconf):\n"
+    << "#no-resolvconf=1\n";
+
 #endif
 #else
   f << "bind=127.0.0.1:53\n";
