From: Jeff Becker <jeff@i2p.rocks>
Date: Sun, 15 Dec 2019 11:59:26 -0500
Subject: dont fail on 0 sized data fed to json parser

---
 llarp/util/json.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/llarp/util/json.cpp b/llarp/util/json.cpp
index 9c9358c..1f0fd22 100644
--- a/llarp/util/json.cpp
+++ b/llarp/util/json.cpp
@@ -23,6 +23,8 @@ namespace llarp
       bool
       FeedData(const char* buf, size_t sz) override
       {
+        if(sz == 0)
+          return true;
         if(m_Offset + sz > m_Buf.size() - 1)
           return false;
         std::copy(buf, buf + sz, m_Buf.begin());
